network/test/middleware_test.go

```diff
@@ -164,6 +164,9 @@ func (m *MiddlewareTestSuite) createOverlay() *mocknetwork.Overlay {
 	overlay := &mocknetwork.Overlay{}
 	overlay.On("Identities").Maybe().Return(m.getIds, nil)
 	overlay.On("Topology").Maybe().Return(m.getIds, nil)
+	// we're only dealing with staked IDs in this test, so queries based on PeerIDs should not succeeded
+	// (they will occur on the Spoofing test)
+	overlay.On("Identity").Return(nil, false)
```

Comment:
Am confused 😕 , shouldn't it be the opposite that since all nodes are staked in the test, the call made by the TopicValidator to overlay.Identity should always succeed?

Reply:
oh, indeed  - I thought this code path only triggered on unstaked nodes, but the topic validator is the one that's going to fire it.